cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -501,7 +508,16 @@ void SparsePoseGraph::RunOptimization() {
 std::vector<std::vector<mapping::TrajectoryNode>>
 SparsePoseGraph::GetTrajectoryNodes() {
   common::MutexLocker locker(&mutex_);
-  return trajectory_nodes_.data();
+  std::vector<std::vector<mapping::TrajectoryNode>> nodes;
+  for (int trajectory_id : trajectory_nodes_.trajectory_ids()) {
+    nodes.resize(trajectory_id + 1);
+    // TODO(cschuet): Rewrite this to code to allow for holes in the trajectory.
```

Comment:
I think it does already. The `resize` default constructs and we treat `nullptr` constant data as trimmed nodes.

Reply:
Done.